### ISSUE TITLE
Very Deep Convolutional Networks for Large-Scale Image Recognition

### ISSUE THREAD

**Issue by caerang** [caerang/reading-papers#1]
# Summary
- 대규모 이미지 인식 문제에서 CNN 기반 네트워크가 깊어질 수록 정확도에 어떤 영향을 미치는지 연구
- 작은 3x3 convolution filter를 16-19레이어 깊이만 쌓아도 성능 향상이 많이 되는 것을 실험으로 증명
- 2014년 ImageNet Challenge에서 1(Localization), 2(Classification)등을 달성
- 일반화 성능이 우수함을 실험으로 증명

# Introduction
- 논문이 발표되던 시기(2014년)에 대규모 공개 데이터와 고속 계산이 가능한 GPU 사용이 가능해 지면서 Convolutional Networks가 대규모 이미지/비디오 인식에서 큰 성공을 보이기 시작함
- ILSVRC에서 최고 성능을 달성하기 위해 ConvNet을 기본으로 사용하기 시작하면서 고차원의 얕은 특징 인코딩부터 깊은 ConvNet까지 다양한 시도가 이루어짐(2012년 Deep Learning 기반의 AlexNet이 최고 성능을 보임)
- 네트워크를 구성할 때 ConvNet이 기본으로 사용되면서 성능을 개선하기 위한 다양한 연구가 진행
    - 첫 convolution layer에 더 작은 receptive window 를 적용하고 stride도 작게 적용
- 본 논문은 ConvNet에서 depth(layer를 더 깊게 쌓는 방법)에 관한 연구를 진행 (모든 layer에서 3x3의 작은 convolution layer를 사용해서 네트워크를 깊이 만드는 것이 전체 성능 향상 시킴을 보임)

# ConvNet Configurations

## Architecture

![vgg_architectures](https://github.com/caerang/reading-papers/assets/11883824/a86b05b6-7e3d-42c6-aa63-9b8adf6d066d)

Preprocessing: subtracting the mean RGB value

모든 히든 레이어에 ReLU 활성화 함수 사용

하나의 네트워크를 제외하고 모든 네트워크에서 LRN 정규화를 사용하지 않았는데, ILSVRC 데이터에서 LRN 정규화(AlexNet에서 사용한 방법)가 성능 향상에 도움이 되지 않음을 확인(오히려 메모리 사용량과 계산 시간 증가의 원인이 됨).

이전 ILSVRC 대회에서 좋은 성능을 보여준 네트워크와 차이점은 VGG는 상대적으로 아주 작은 3x3 conv.layer를 사용했다는 점(이전 네트워크는 첫 conv.layer로 11x11s2, 7x7s2 등을 사용)

3x3 conv.layer 두 개를 겹쳐서 쌓으면 한 개의 5x5 conv.layer의 효과를 가짐 3개의 3x3 conv.layer를 겹쳐서 쌓으면 한 개의 7x7 conv.layer의(receptive field, 수용 영역, 수용장과 동일한) 효과를 가짐

> [!NOTE]
> **한 개의 7x7 conv.layer를 사용하는 대신 3 개의 3x3 conv.layer를 겹쳐서 쌓는 것의 장점은 무엇일까?**
> - 3 개의 비선형 활성화 함수를 사용할 수 있어 결정 함수의 구분력이 좋게 만든다.(공간을 더 비선형 공간으로 변환할 수 있다.)
> - 학습 파라미터의 수를 줄일 수 있다. (이는 7x7 conv.filters에 regularization을 적용해서 3x3 conv.filter로 분해(비선형성을 강화시키면서 )를 강제하는 것으로 해석할 수 있음)

1x1 conv.layers는 conv.layer의 수용영역(receptive field)에 영향을 미치지 않으면서 결정 함수(decision function)에 비선형성을 증가시키는 방법

Goodfellow 은 2014년에 11개의 가중치 레이어로 이루어진 deep ConvNets로 숫자 인식에 적용했고, 깊이가 깊어질 수록 성능이 좋아짐을 보임

GoogLeNet도 VGG와는 독립적으로 개발이 진행(ILSVRC-2014 에서 순위 권 성능)

VGG가 GoogLeNet보다 구조가 단순

## Training

multinomial logistic regression objective

mini-batch gradient descent with momentum: batch size 256, momentum 0.9

weight decay(L2 penalty numtiplier set to 5*10-4

dropout regularisation for the first two fully-connected layer (dropout ratio 0.5)

learning rate: 10-2, 학습 정시 시에는 3배 감소돼었음

74epochs에 학습 정지(370K iterations)

작은 conv 필터 크기와 대단히 깊은 네트워크가 암묵적으로 regularisation하는 역할을 수행함

특정 레이어를 사전에 초기화 함

네트워크가 깊어질 때 가중치의 불안정으로 학습이 되지 않을 수 있기 때문에 가중치를 초기화 하는 것은 중요함
